src/game/client/draw/w3dmodeldraw.cpp

```diff
@@ -1154,3 +1178,1890 @@ void W3DModelDrawModuleData::Parse_Condition_State(INI *ini, void *instance, voi
+                    m_whichAnimInCurState = Get_Client_Random_Value(0, count - 1);
+                }
+            } else {
+                m_whichAnimInCurState = Get_Client_Random_Value(0, count - 1);
+            }
+
+            HAnimClass *anim = m_curState->m_animations[m_whichAnimInCurState].Get_Anim_Handle();
+
+            if (m_renderObject != nullptr && anim != nullptr) {
+                int frame = 0;
+
+                if (m_curState->m_mode == RenderObjClass::ANIM_MODE_ONCE_BACKWARDS
+                    || m_curState->m_mode == RenderObjClass::ANIM_MODE_LOOP_BACKWARDS) {
+                    frame = anim->Get_Num_Frames() - 1;
+                }
+
+                if (Test_Animation_Flag(m_curState->m_flags, RANDOMSTART)) {
+                    frame = Get_Client_Random_Value(0, anim->Get_Num_Frames() - 1);
+                } else if (Test_Animation_Flag(m_curState->m_flags, START_FRAME_FIRST)) {
+                    frame = 0;
+                } else if (Test_Animation_Flag(m_curState->m_flags, START_FRAME_LAST)) {
+                    frame = anim->Get_Num_Frames() - 1;
+                } else if (Maintain_Frame_Across_States(m_curState->m_flags) && prev_state != nullptr
+                    && prev_state != m_curState && Maintain_Frame_Across_States(prev_state->m_flags)
+                    && Maintain_Frame_Across_States_2(m_curState->m_flags, prev_state->m_flags) && f >= 0.0f) {
+                    frame = GameMath::Fast_To_Int_Truncate(anim->Get_Num_Frames() * f - 1.0f);
+                }
+
+                m_renderObject->Set_Animation(anim, frame, m_curState->m_mode);
+                anim->Release_Ref();
+
+                if (m_renderObject->Class_ID() == RenderObjClass::CLASSID_HLOD) {
+                    HLodClass *hlod = static_cast<HLodClass *>(m_renderObject);
+                    hlod->Set_Animation_Frame_Rate_Multiplier(Get_Client_Random_Value_Real(
+                        m_curState->m_animationSpeedFactorMin, m_curState->m_animationSpeedFactorMax));
+                }
+            }
+        }
+    }
+}
+
+bool W3DModelDraw::Set_Cur_Anim_Duration_In_Msec(float desired_duration_in_msec)
+{
+    if (m_renderObject == nullptr || m_renderObject->Class_ID() != RenderObjClass::CLASSID_HLOD) {
+        return false;
+    }
+
+    HLodClass *hlod = static_cast<HLodClass *>(m_renderObject);
+    HAnimClass *anim = hlod->Peek_Animation();
+
+    if (anim == nullptr) {
+        return false;
+    }
+
+    float f = (anim->Get_Num_Frames() * 1000.0f) / anim->Get_Frame_Rate();
+
+    if (f <= 0.0f || desired_duration_in_msec <= 0.0f) {
+        return false;
+    }
+
+    hlod->Set_Animation_Frame_Rate_Multiplier(f / desired_duration_in_msec);
+    return true;
+}
+
+float W3DModelDraw::Get_Cur_Anim_Distance_Covered() const
+{
+    if (m_curState == nullptr || m_whichAnimInCurState < 0) {
+        return 0.0f;
+    }
+
+#ifdef GAME_DEBUG_STRUCTS
+    if (s_skateDistanceOverride = 0.0f) {

```

Comment:
This condition assigns value, instead of compare.

Reply:
fixed.